src/System.Windows.Forms/src/System/Windows/Forms/DateTimePicker.DateTimePickerAccessibleObject.cs

```diff
@@ -114,8 +114,26 @@ internal override object GetPropertyValue(UiaCore.UIA propertyID)
             internal override bool IsPatternSupported(UiaCore.UIA patternId)
                 => (patternId == UiaCore.UIA.TogglePatternId && ((DateTimePicker)Owner).ShowCheckBox) ||
                     patternId == UiaCore.UIA.ExpandCollapsePatternId ||
+                    patternId == UiaCore.UIA.LegacyIAccessiblePatternId ||
                     base.IsPatternSupported(patternId);
```

Comment:
Could you please refactor this using switch expression? @Tanya-Solyanik what do you think about readability of switch expression in this case?
```suggestion
                => patternId switch
                {
                    UiaCore.UIA.TogglePatternId when ((DateTimePicker)Owner).ShowCheckBox: true,
                    UiaCore.UIA.ExpandCollapsePatternId: true,
                    UiaCore.UIA.LegacyIAccessiblePatternId: true,
                    _ => base.IsPatternSupported(patternId)
                };
```

Reply:
I prefer the switch expression in terms of readability, the general guidance is to modernize the code base as we go along. 
In this particular case, `IsPatternSupported` is implemented as a `switch` statement/expression in other classes, so additionally it makes the code more consistent.

Reply:
It looks better, thanks. Reworked.

Reply:
I also reworked `DefaultAction` property using the match pattern. But seems I can't do this for `DoDefaultAction` method due to default case where I can't call some function or return something because the method has void return-type. So situation the same as when use a ternary conditional.

Reply:
Don't mind, reworked `DoDefaultAction` using switch.